File: iossdkdemo/SDKDemo/APIControllers/GetPanelsViewController.m
//
//  GetPanelsViewController.m
//  SDKDemo
//
//  Created by OnePoint Global on 04/10/16.
//  Copyright © 2016 opg. All rights reserved.
//

#import "GetPanelsViewController.h"
#import <OPGSDK/OPGSDK.h>
#import <OPGSDK/OPGPanel.h>
#import <OPGSDK/OPGPanellistPanel.h>
@interface GetPanelsViewController ()

@property(strong, nonatomic) NSArray* panelsObjectList;
@property(strong, nonatomic) NSMutableArray* panelsIDList;

@end

@implementation GetPanelsViewController

@synthesize panelsArray;


-(NSMutableArray*) getPanelIDList : (NSArray*)panels
{
    NSMutableArray *listOfPanelID = [NSMutableArray new];
    for (OPGPanel *panel in panels)
    {
        [listOfPanelID addObject:panel.panelName];
    }
    return listOfPanelID;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    self.tableView.separatorColor = [UIColor orangeColor];
    // Uncomment the following line to preserve selection between presentations.
    // self.clearsSelectionOnViewWillAppear = NO;
    
    // Uncomment the following line to display an Edit button in the navigation bar for this view controller.
    // self.navigationItem.rightBarButtonItem = self.editButtonItem;
    
    OPGSDK* sdk = [OPGSDK new];                   // Creating OPGSDK instance
    NSError *error;
    OPGPanellistPanel *panellistPanelObj = [sdk getPanellistPanel:&error];
    self.panelsObjectList = panellistPanelObj.panelsArray;
    self.panelsArray = [self getPanelIDList:self.panelsObjectList];
    if (self.panelsArray.count<=0)
    {
        UIAlertController * alert = [UIAlertController
                                     alertControllerWithTitle:@"OPGSDKv2.0"
                                     message:@"No Panels!"
                                     preferredStyle:UIAlertControllerStyleAlert];
        
        UIAlertAction* okButton = [UIAlertAction
                                   actionWithTitle:@"OK"
                                   style:UIAlertActionStyleDefault
                                   handler:^(UIAlertAction * action) {
                                       //Handle your yes please button action here
                                   }];
        
        [alert addAction:okButton];
    }
    [self.tableView reloadData];

}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - Table view data source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.panelsArray.count;
}


- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"myCell"];
    
    if (cell == nil) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleSubtitle reuseIdentifier:@"myCell"];
        cell.selectionStyle = UITableViewCellSelectionStyleNone;
    }
    
    cell.textLabel.text = [self.panelsArray objectAtIndex:indexPath.row];
    return cell;
}


/*
 // Override to support conditional editing of the table view.
 - (BOOL)tableView:(UITableView *)tableView canEditRowAtIndexPath:(NSIndexPath *)indexPath {
 // Return NO if you do not want the specified item to be editable.
 return YES;
 }
 */

/*
 // Override to support editing the table view.
 - (void)tableView:(UITableView *)tableView commitEditingStyle:(UITableViewCellEditingStyle)editingStyle forRowAtIndexPath:(NSIndexPath *)indexPath {
 if (editingStyle == UITableViewCellEditingStyleDelete) {
 // Delete the row from the data source
 [tableView deleteRowsAtIndexPaths:@[indexPath] withRowAnimation:UITableViewRowAnimationFade];
 } else if (editingStyle == UITableViewCellEditingStyleInsert) {
 // Create a new instance of the appropriate class, insert it into the array, and add a new row to the table view
 }
 }
 */

/*
 // Override to support rearranging the table view.
 - (void)tableView:(UITableView *)tableView moveRowAtIndexPath:(NSIndexPath *)fromIndexPath toIndexPath:(NSIndexPath *)toIndexPath {
 }
 */

/*
 // Override to support conditional rearranging of the table view.
 - (BOOL)tableView:(UITableView *)tableView canMoveRowAtIndexPath:(NSIndexPath *)indexPath {
 // Return NO if you do not want the item to be re-orderable.
 return YES;
 }
 */

/*
 #pragma mark - Navigation
 
 // In a storyboard-based application, you will often want to do a little preparation before navigation
 - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
 // Get the new view controller using [segue destinationViewController].
 // Pass the selected object to the new view controller.
 }
 */

@end

File: iossdkdemo/SDKDemo-Swift/APIControllers-Swift/GetPanelsViewController.swift
//
//  GetPanelsViewController.swift
//  SDKDemoApp
//
//  Created by OnePoint Global on 04/10/16.
//  Copyright © 2016 opg. All rights reserved.
//

import UIKit

class GetPanelsViewController: UITableViewController
{
    var panelsObjList : NSArray = []
    var panelsIDList : NSArray = []
    var panellistPanelObj : OPGPanellistPanel!
    
    override func viewDidLoad()
    {
        super.viewDidLoad()
        self.tableView.separatorColor = UIColor.orange
        let sdk = OPGSDK()        // Creating OPGSDK instance
        do {
            self.panellistPanelObj = try sdk.getPanellistPanel()
            self.panelsObjList = panellistPanelObj.panelsArray as NSArray       // Getting list of Panels assigned to user
            self.panelsIDList = self.getPanelsIDList(self.panelsObjList)
        }
        catch{
            print("Get Panels not working")         /* @"Error Occured. Contact Support!" */
            
        }
        // Do any additional setup after loading the view, typically from a nib.
    }
    
    func getPanelsIDList(_ obj : NSArray) -> NSArray
    {
        let panelsIDArray : NSMutableArray = []
        for panel in obj as! [OPGPanel]
        {
            panelsIDArray.add(panel.panelName)
        }
        return panelsIDArray
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return self.panelsIDList.count
    }
    
    override func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell
    {
        var cell = tableView.dequeueReusableCell(withIdentifier: "CELL") as UITableViewCell!
        if (cell == nil) {
            cell = UITableViewCell(style:.default, reuseIdentifier: "CELL")
        }
        // setup cell without force unwrapping it
        cell?.textLabel!.text = self.panelsIDList[(indexPath as NSIndexPath).row] as? String
        return cell!
        
        
    }

}
